src/rules/parser.go

```diff
@@ -558,3 +558,37 @@ func (p *parser) checkForVariableInPattern(name string, pattern ir.Node, verifie
 
 	return found
 }
+
+func (p *parser) filterByPattern(name string, pattern ir.Node, verifiedVars map[string]struct{}) bool {
```

Comment:
Let's add the tests? It's a very fragile place

Reply:
Done